python/pyiceberg/table/__init__.py

```diff
@@ -934,206 +961,608 @@ def case_sensitive(self, case_sensitive: bool) -> UpdateSchema:
         return self
 
     def add_column(
-        self, name: str, type_var: IcebergType, doc: Optional[str] = None, parent: Optional[str] = None, required: bool = False
+        self, path: Union[str, Tuple[str, ...]], field_type: IcebergType, doc: Optional[str] = None, required: bool = False
     ) -> UpdateSchema:
         """Add a new column to a nested struct or Add a new top-level column.
 
         Args:
-            name: Name for the new column.
-            type_var: Type for the new column.
+            path: Name for the new column.
+            field_type: Type for the new column.
             doc: Documentation string for the new column.
-            parent: Name of the parent struct to the column will be added to.
             required: Whether the new column is required.
 
         Returns:
-            This for method chaining
+            This for method chaining.
         """
-        if "." in name:
-            raise ValueError(f"Cannot add column with ambiguous name: {name}")
+        path = (path,) if isinstance(path, str) else path
+
+        if "." in path[-1]:
+            raise ValueError(f"Cannot add column with ambiguous name: {path[-1]}, provide a tuple instead")
```

Comment:
This isn't quite correct. It's actually fine for the name to contain a `.` if the path was a tuple. If it was a tuple, then we know exactly what part is the parent. So this check is only needed if path is a str:

```python
if isinstance(path, str):
    if '.' in path:
        raise ValueError(...)
    path = (path,)

...
```